pronotepy/dataClasses.py

```diff
@@ -492,20 +492,26 @@ def __init__(self, client, json_dict):
         self.end: datetime.datetime = self.start.replace(hour=end_time.hour, minute=end_time.minute)
 
         # get additional information about the lesson
-        self.teacher_name: Optional[str] = None
-        self.classroom: Optional[str] = None
-        self.group_name: Optional[str] = None
+        self.teacher_names: Optional[List[str]] = []
+        self.classrooms: Optional[List[str]] = []
+        self.group_names: Optional[List[str]] = []
         self.subject: Optional[Subject] = None
 
         if 'ListeContenus' not in json_dict:
             raise ParsingError("Error while parsing for lesson details", json_dict, ("ListeContenus", "V"))
 
         for d in json_dict['ListeContenus']['V']:
-            try:
-                self.__setattr__(self.__class__.transformers[d['G']][0], self.__class__.transformers[d['G']][1](d))
-            except KeyError:
-                pass
-
+            if not 'G' in d: continue
+            if d['G'] == 16: self.subject = Subject(d)
+            if d['G'] == 3: self.teacher_names.append(d['L'])
+            if d['G'] == 17: self.classrooms.append(d['L'])
+            if d['G'] == 2: self.group_names.append(d['L'])
```

Comment:
Super minor nitpick. LGTM
```suggestion
            if not 'G' in d: continue
            elif d['G'] == 16: self.subject = Subject(d)
            elif d['G'] == 3: self.teacher_names.append(d['L'])
            elif d['G'] == 17: self.classrooms.append(d['L'])
            elif d['G'] == 2: self.group_names.append(d['L'])
```